server/base/src/main/java/org/apache/accumulo/server/metadata/iterators/SetEqualityIterator.java

```diff
@@ -186,4 +187,49 @@ public static <T> Condition createCondition(Collection<T> set, Function<T,byte[]
     return new Condition(family, EMPTY).setValue(encode((Set<T>) set, encoder)).setIterators(is);
   }
 
+  private static <K,T> Set<T> extractValuesFromMap(Map<K,T> map) {
+    return new HashSet<>(map.values());
+  }
+
+  private static <K,T> Set<K> extractKeysFromMap(Map<K,T> map) {
+    return new HashSet<>(map.keySet());
+  }
+
+  public static <K,T> byte[] encodeMapValues(Map<K,T> map, Function<T,byte[]> encoder) {
+    Set<T> valuesSet = extractValuesFromMap(map);
+    return encode(valuesSet, encoder);
+  }
+
+  public static <K,T> byte[] encodeMapKeys(Map<K,T> map, Function<K,byte[]> encoder) {
+    Set<K> keysSet = extractKeysFromMap(map);
+    return encode(keysSet, encoder);
+  }
+
+  public static <K,T> Condition createMapValuesCondition(Map<K,T> map, Function<T,byte[]> encoder,
+      Text family) {
+    Preconditions.checkArgument(map instanceof Map);
+    IteratorSetting is = new IteratorSetting(ConditionalTabletMutatorImpl.INITIAL_ITERATOR_PRIO,
+        SetEqualityIterator.class);
+    return new Condition(family, EMPTY).setValue(encodeMapValues(map, encoder)).setIterators(is);
+  }
+
+  public static <K,T> Condition createMapKeysCondition(Map<K,T> map, Function<K,byte[]> encoder,
+      Text family) {
```

Comment:
We could possibly change this to encode the keys and values for the map with a method like the following.

```suggestion
  public static <K,T> Condition createMapKeysCondition(Map<K,T> map, Function<Entry<K,T>,Entry<byte[],byte[]>> encoder,
      Text family) {
```

This could then call entry set on the map and then encode the entry set using existing methods.  Would also need to add an option to the iterator to encode the values on the server side.